File: swipe_practice3/app/src/main/java/com/example/swipe_practice/MainActivity.java
package com.example.swipe_practice;

import androidx.appcompat.app.AppCompatActivity;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.content.Context;
import android.graphics.Color;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    SwipeRefreshLayout swipeRefreshLayout;
    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        swipeRefreshLayout=findViewById(R.id.swipe);
        textView=findViewById(R.id.textView);
        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                swipeRefreshLayout.setRefreshing(false);

                boolean connection=isNetworkAvailable();
                if(connection){
                    textView.setText("internet connect");
                    textView.setTextColor(Color.GREEN);
                }
                else{
                    textView.setTextColor(Color.RED);
                    textView.setText("not connected");
                }
            }
        });
        swipeRefreshLayout.setColorSchemeColors(Color.YELLOW);

    }

    public Boolean isNetworkAvailable(){
        ConnectivityManager connectivityManager=(ConnectivityManager) this.getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo=connectivityManager.getActiveNetworkInfo();
        return networkInfo !=null;
    }
}

File: Udemy/app/src/main/java/com/example/udemy/MainActivity.kt
package com.example.udemy

import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.databinding.DataBindingUtil
import com.example.udemy.databinding.ActivityMainBinding
import kotlin.random.Random

class MainActivity : AppCompatActivity() {
    private lateinit var databinding:ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        databinding=DataBindingUtil.setContentView(this,R.layout.activity_main)

      /* val button=databinding.save
         button.setOnClickListener(View.OnClickListener {
            greeting()
        })

       */
        // ....................OR.............................


        databinding.save.setOnClickListener(View.OnClickListener {
            greeting()
        })

        databinding.progress.setOnClickListener(View.OnClickListener {
            if (databinding.progressBar.visibility==View.GONE){
                databinding.progressBar.visibility=View.VISIBLE
                databinding.progress.text = "stop"
                databinding.progress.setBackgroundColor(Color.RED)
            }
            else  if(databinding.progressBar.visibility==View.VISIBLE){
                databinding.progressBar.visibility=View.GONE
                databinding.progress.text="start"
                databinding.progress.setBackgroundColor(Color.GREEN)
            }

        })

        databinding.student=getStudent()
    }

    private fun greeting(){
        val text=databinding.textView
        //var ran:Random= Random
       // var ran1=Random.nextInt(255)

        text.text = "Bonjour....you clicked!!!!"
        text.setTextColor(Color.rgb(Random.nextInt(255),Random.nextInt(255),Random.nextInt(255)))
        databinding.save.setBackgroundColor(Color.rgb(Random.nextInt(255),Random.nextInt(255),Random.nextInt(255)))
        text.textSize= 32.0F


    }

    private fun getStudent():Student{
        return Student(1, "JAVIER","javier.harvard.ac.soc")
    }
}

class Student(i:Int,n:String,e:String) {
    val id=i
    val name=n
    val email=e



}



